querybook/webapp/components/AIAssistant/AICommandInput.tsx

```diff
@@ -31,52 +31,61 @@ interface AICommandInputProps {
 
 export const AICommandInput: React.FC<AICommandInputProps> = forwardRef(
     (
-        {
-            commands = [],
-            placeholder,
-            running,
-            onCommandChange,
-            onSubmit,
-            cancelGeneration,
-        },
+        { commands = [], running, onCommandChange, onSubmit, cancelGeneration },
         ref
     ) => {
         const textareaRef = useForwardedRef<IResizableTextareaHandles>(ref);
         const anchorRef = useRef<HTMLDivElement>(null);
-        const [commandName, setCommandName] = useState('');
+        const [command, setCommand] = useState<IQueryCellCommand>();
         const [commandValue, setCommandValue] = useState('');
 
         const [filteredCommands, setFilteredCommands] = useState([]);
         const [currentCommandItemIndex, setCurrentCommandItemIndex] =
             useState(0);
 
         useEffect(() => {
-            const command = commands.find((cmd) => cmd.name === commandName);
             onCommandChange(command, commandValue);
-        }, [commandName, commandValue]);
+        }, [command, commandValue]);
 
-        const handleChange = (value: string) => {
-            if (commandName || !value.startsWith('/')) {
-                setCommandValue(value);
+        const setNewCommand = useCallback(
+            (newCommand: IQueryCellCommand) => {
+                setCommand(newCommand);
                 setFilteredCommands([]);
-                return;
-            }
+                setCommandValue('');
+            },
+            [setCommand, setCommandValue, setFilteredCommands]
+        );
 
-            // when value starts with "/"
-            const prefix = value.slice(1).toLocaleLowerCase(); // trim the leading '/'
+        const handleChange = useCallback(
+            (value: string) => {
+                if (command || !value.startsWith('/')) {
```

Comment:
i guess you will need to trim the value here to extract the /, actually i think its fine, let's just not trim